### PR TITLE
Topic/iterators

### PR DESCRIPTION
**Pull request by godunko**

### CODE REVIEW

File: source/text/vss-strings.ads [hunk 1]
```diff
@@ -98,6 +98,241 @@ package VSS.Strings is
...
+
+   procedure Clear (Self : in out Virtual_String'Class);
+   --  Remove all data.
+
+   procedure Append
+     (Self : in out Virtual_String'Class;
+      Item : VSS.Characters.Virtual_Character);
+   --  Append given abstract character to the end of the string.
+
+   --  function Append
```
Comment (reznikmm) [inline source/text/vss-strings.ads:123]: Why do we need Append functions if we have "&"?

File: source/text/vss-strings.ads [hunk 2]
```diff
@@ -98,6 +98,241 @@ package VSS.Strings is
...
+   --
+   --  procedure Append
+   --    (Self : in out Virtual_String'Class;
+   --     Item : Virtual_String'Class);
+   --  --  Append given string to the end of the string.
+   --
+   --  function Append
+   --    (Self : Virtual_String'Class;
+   --     Item : Virtual_String'Class) return Virtual_String;
+   --  --  Append given string to the end of the string and returns result.
```
Comment (reznikmm) [inline source/text/vss-strings.ads:137]: see above

File: source/text/vss-strings.ads [hunk 3]
```diff
@@ -98,6 +98,241 @@ package VSS.Strings is
...
+   --     Position : VSS.Strings.Cursors.Abstract_Character_Cursor'Class;
+   --     Item     : VSS.Characters.Virtual_Character);
+   --  procedure Insert
+   --    (Self     : in out Virtual_String'Class;
+   --     Position : VSS.Strings.Cursors.Abstract_Character_Cursor'Class;
+   --     Item     : Virtual_String'Class);
+   --  procedure Insert
+   --    (Self     : in out Virtual_String'Class;
+   --     Position : VSS.Strings.Cursors.Abstract_Segment_Cursor'Class;
+   --     Item     : VSS.Characters.Virtual_Character);
```
Comment (reznikmm) [inline source/text/vss-strings.ads:160]: Do we really need `Insert` for segment cursors? Where do inserted character go? After the segment end or after the first character of segment beginning?

Could be not clear for the user.

**Review by reznikmm — CHANGES_REQUESTED** [review]

File: source/text/vss-strings.ads (hunk 1 above)
Comment (godunko) [reply source/text/vss-strings.ads:123]: Why not? It is alternative name for completeness.

File: source/text/vss-strings.ads (hunk 3 above)
Comment (godunko) [reply source/text/vss-strings.ads:152]: Insert operations insert data at position pointing by the cursor and moves existing data after inserted one. It is not important is it single character element or segment element.

File: source/text/vss-strings.ads (hunk 1 above)
Comment (reznikmm) [reply source/text/vss-strings.ads:128]: * It makes API more complex, 
* Append isn't a good name for it, may be Concatenate is better, but I would avoid it at all

**Review by reznikmm — APPROVED** [review]

**Pull request merged**